specs/online_order_spec.rb

```diff
@@ -1,90 +1,161 @@
 
-      # Instatiate your OnlineOrder here
-      # online_order =
-      # online_order.must_be_kind_of Grocery::Order
+      online_order = Grocery::OnlineOrder.new(1325, products, customer, :complete)
+
+      online_order.must_be_kind_of Grocery::Order
     end
 
     it "Can access Customer object" do
-      # TODO: Your test code here!
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      customer = Grocery::Customer.new(1, "leonard.rogahn@hagenes.org", {:street=>"71596 Eden Route", :city=>"Connellymouth", :state=>"LA", :zip_code=>"98872-9105"})
+
+      online_order = Grocery::OnlineOrder.new(1325, products, customer, :complete)
+
+      online_order.customer.must_be_instance_of Grocery::Customer
     end
 
     it "Can access the online order status" do
-      # TODO: Your test code here!
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      customer = Grocery::Customer.new(1, "leonard.rogahn@hagenes.org", {:street=>"71596 Eden Route", :city=>"Connellymouth", :state=>"LA", :zip_code=>"98872-9105"})
+
+      online_order = Grocery::OnlineOrder.new(1325, products, customer, :complete)
+
+      online_order.status.must_equal :complete
     end
   end
 
   describe "#total" do
     it "Adds a shipping fee" do
-      # TODO: Your test code here!
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      customer = Grocery::Customer.new(1, "leonard.rogahn@hagenes.org", {:street=>"71596 Eden Route", :city=>"Connellymouth", :state=>"LA", :zip_code=>"98872-9105"})
+      online_order = Grocery::OnlineOrder.new(1325, products, customer, :complete)
+
+      sum = products.values.inject(0, :+)
+      cents_total_fee = (sum + (sum * 0.075))*100.round + 1000
+      total_with_money = online_order.total
+
+      total_with_money.must_equal Money.new(cents_total_fee, "USD")
     end
 
     it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+      products = {}
+      customer = Grocery::Customer.new(1, "leonard.rogahn@hagenes.org", {:street=>"71596 Eden Route", :city=>"Connellymouth", :state=>"LA", :zip_code=>"98872-9105"})
+      online_order = Grocery::OnlineOrder.new(1325, products, customer, :complete)
+
+      total_with_money = online_order.total
+
+      total_with_money.must_equal Money.new(0, "USD")
     end
   end
 
   describe "#add_product" do
-    it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
+    it "Permits action for pending and paid satuses" do
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      customer = Grocery::Customer.new(1, "leonard.rogahn@hagenes.org", {:street=>"71596 Eden Route", :city=>"Connellymouth", :state=>"LA", :zip_code=>"98872-9105"})
+
+      permitted_statuses = [:pending, :paid]
+      permitted_statuses.each do |status|
+        online_order = Grocery::OnlineOrder.new(1325, products, customer, status)
+
+        online_order.add_product("salad", 4.25).wont_be_nil
```

Comment:
Just like the `Order` class the `add_product` method should return `true` if it works and `false` if it doesn't.